execution/stagedsync/stage_snapshots.go

```diff
@@ -280,6 +281,7 @@ func DownloadAndIndexSnapshotsIfNeed(s *StageState, ctx context.Context, tx kv.R
 		agg,
 		tx,
 		cfg.blockReader,
+		rawdbv3.TxNums.WithCustomReadTxNumFunc(freezeblocks.TxBlockIndexFromBlockReader(ctx, cfg.blockReader)),
```

Comment:
@sudeepdino008 is it right way to create txnum reader?

Reply:
not anymore; use `cfg.blockReader.TxNumReader(ctx)`